src/main/java/com/iridium/iridiumskyblock/gui/BlockValueGUI.java

```diff
@@ -41,20 +43,27 @@ public void addContent(Inventory inventory) {
 
         InventoryUtils.fillInventory(inventory, getNoItemGUI().background);
 
+        inventory.setItem(inventory.getSize() - 3, ItemStackUtils.makeItem(IridiumSkyblock.getInstance().getInventories().nextPage));
+        inventory.setItem(inventory.getSize() - 7, ItemStackUtils.makeItem(IridiumSkyblock.getInstance().getInventories().previousPage));
+
         if (guiType == BlockValueType.BLOCK) {
-            for (Map.Entry<XMaterial, ValuableBlock> valuableBlock : IridiumSkyblock.getInstance().getBlockValues().blockValues.entrySet()) {
-                XMaterial material = valuableBlock.getKey();
-                ValuableBlock blockInfo = valuableBlock.getValue();
-                ItemStack blockItem = ItemStackUtils.makeItem(material, 1, StringUtils.color(blockInfo.name), getColoredValueLore(blockInfo.value));
-                inventory.setItem(blockInfo.slot, blockItem);
-            }
+            IridiumSkyblock.getInstance().getBlockValues().blockValues.entrySet().stream()
+                    .filter(valuableBlockEntry -> (page == 1 ? valuableBlockEntry.getValue().page <= 1 : valuableBlockEntry.getValue().page == page))
+                    .forEachOrdered(valuableBlock -> {
+                        XMaterial material = valuableBlock.getKey();
+                        ValuableBlock blockInfo = valuableBlock.getValue();
+                        ItemStack blockItem = ItemStackUtils.makeItem(material, 1, StringUtils.color(blockInfo.name), getColoredValueLore(blockInfo.value));
+                        inventory.setItem(blockInfo.slot, blockItem);
+                    });
+
         } else if (guiType == BlockValueType.SPAWNER) {
-            for (Map.Entry<EntityType, ValuableBlock> valuableSpawner : IridiumSkyblock.getInstance().getBlockValues().spawnerValues.entrySet()) {
-                EntityType spawnerType = valuableSpawner.getKey();
-                ValuableBlock spawnerInfo = valuableSpawner.getValue();
-                ItemStack spawnerItem = ItemStackUtils.makeItem(XMaterial.SPAWNER, 1, StringUtils.color(spawnerInfo.name), getColoredValueLore(spawnerInfo.value));
-                inventory.setItem(spawnerInfo.slot, spawnerItem);
-            }
+            IridiumSkyblock.getInstance().getBlockValues().spawnerValues.entrySet().stream()
+                    .filter(valuableSpawnerEntry -> (page == 1 ? valuableSpawnerEntry.getValue().page <= 1 : valuableSpawnerEntry.getValue().page == page))
+                    .forEachOrdered(valuableSpawner -> {
```

Comment:
You should probably use the map method here. It's up to you but I can help you with it if you want

Reply:
how :/

Reply:
we must have entryset and map method will get only the value without key

Reply:
Can I do it for you?

Reply:
ofc pls <3

Reply:
Nvm